src/Optimize/OptimizeBase.h

```diff
@@ -26,7 +27,17 @@ template<class T = double>
 class CostFunctionBase
 {
 public:
-  typedef T Return_t;
+  //typedef T Return_t;
+
+  //#ifdef QMC_COMPLEX
+  //typedef std::complex<T> Return_ct;
+  //#else
+  //typedef T Return_ct;
+  //#endif
+
+  typedef qmcplusplus::QMCTraits::ValueType Return_ct;
```

Comment:
This is a naming issue. Return_ct is real type when building the real code.
Better to use Return_t for ValueType and Return_rt for RealType.

Reply:
I have fixed this in the latest commit.